File: NodeJS/RxConsole/service.js

// const https = require('https');
// const URL = 'https://reqres.in/api/users';
//
// https.get(`${URL}?page=1&per_page=50`, res => {

//     res.setEncoding("utf8");
//     let body = "";
//     res.on("data", data => {
//         body += data;
//     });

//     res.on('end', () => {
//         body = JSON.parse(body);
//         console.log(body);
//       });      
// });

//
// Using axios dependency 
//
const axios = require("axios");
const URL = 'https://reqres.in/api/users';

const getUsersAsync = async () => {
  try {
    const response = await axios.get(`${URL}?pages=1&per_page=50`);
    const users = response.data;
    return users;
  } catch (error) {
    console.log(error);
  }
};

// exports.getUsersAsync = getUsersAsync;
module.exports = {
  getUsersAsync
}

File: CSharp/RxConsole/Service.cs
using System.Net.Http;
using Newtonsoft.Json;
using System.Threading.Tasks;
using System;
using System.Reactive.Linq;
using System.ComponentModel;
using System.Reactive.Disposables;

//
// https://reqres.in/
//
namespace CSharpRxConsole
{

    public class Service
    {
        private const string URL = "https://reqres.in/api/users";
        private HttpClient _http;        

        public Service(HttpClient http)
        {
            _http = http;
        }

        // Observable sample
        public IObservable<string> GetObservableSample()
        {
            // Using Javascript !
            //var observable = new Observable((observer) => {
            //    observer.next("Hello Juanlu!");
            //    observer.next("Hello Eva!");
            //    observer.complete();
            //    // observer.next('Bye'); // Undefined. After complete(). Nothing to show !
            //});

            var observable =  Observable.Create((IObserver<string> observer) =>
            {
                observer.OnNext("Hello Juanlu !");
                observer.OnNext("Hello Eva !");                
                observer.OnCompleted();
                // return Disposable.Create(() => Console.WriteLine("Observer has unsubscribed"));
                // or can return an Action like 
                // return () => Console.WriteLine("Observer has unsubscribed");
                return () => { };
            });           

            return observable;
        }

        private async Task<PagedUsers> GetUsersAsync()
        {
            var json = await this._http.GetStringAsync($"{URL}?page=1&per_page=50");
            var users = JsonConvert.DeserializeObject<PagedUsers>(json);

            return users;
        }

        public async Task<IObservable<User>> GetUsersToObservable()
        {
            var data = await GetUsersAsync();

            return data.Users.ToObservable();
        }

        public async Task<IObservable<User>> GetObservableUsers()
        {
            var data = await GetUsersAsync();

            var observable = Observable.Create<User>(observer =>
            {
                foreach (var u in data.Users)
                {
                    observer.OnNext(u);
                }
                observer.OnCompleted();
                return () => { };
            });

            return observable;
        }
    }        
}